api/report_server.thrift

```diff
@@ -93,7 +90,7 @@ typedef list<ReportFilter> ReportFilterList
  * Members of this struct are interpreted in "OR" relation with each other.
```

Comment:
I think this comment is very confusing. Both sentences referring to the members of the struct. Is this correct? 
Why do we need the `v1`? Can we remove that now?

Reply:
As a side note, I am not sure this has the same expressivity as the old version. 

E.g. how do you represent:
```
(filepath == A AND checkerName == B) OR (severity == C and reviewStatus == D)
```

Note that the old format was basically a disjunctive normal form, so we could be sure that every predicate could be translated to that.

Reply:
v1 can only be removed if the command-line client is upgraded to use the same, new, "v2" filtering semantics.

`(filepath == A AND checkerName == B) OR (severity == C and reviewStatus == D)`
This kind of filter was not explicable by any of the clients in the v1 version either, even if the API -- if directly called -- seemed to had supported that.

@gyorb, @csordasmarton rewrote the filtering system.

Reply:
The old `ReportFilter` will be removed.